File: bench/vs-lang/fib.py
# fib.py - Python twin of fib.c / fib.lm / fib.rs, same recursive structure.
# Must print byte-identical stdout to the other twins: "2178309\n".
# Run via CPython (interpreted, no compile step): python3 fib.py


def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    print(fib(32))


if __name__ == "__main__":
    main()

File: bench/vs-lang/fib.rs
// fib.rs - Rust twin of fib.c / fib.lm, same recursive structure.
// Must print byte-identical stdout to the other twins: "2178309\n".

fn fib(n: i64) -> i64 {
    if n < 2 {
        return n;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    println!("{}", fib(32));
}
